src/controller/CHIPDeviceController.cpp

```diff
@@ -2061,7 +2040,16 @@ void DeviceCommissioner::PerformCommissioningStep(DeviceProxy * proxy, Commissio
     }
     break;
     case CommissioningStage::kFindOperational: {
-        CHIP_ERROR err = UpdateDevice(proxy->GetDeviceId());
+        const PeerId peerId              = mFabricInfo->GetPeerIdForNode(proxy->GetDeviceId());
+        OperationalDeviceProxy * session = GetDeviceSession(peerId);
```

Comment:
Does GetDeviceSession create a session if none exists? Does that trigger a lookup of the address if we don't already have it? If so, does that create a double lookup with the LookupPeerAddress called below?

Reply:
Yes, it totally does double-lookup so I had to update Address resolver to handle that (it ignores double lookups with the logic of "I am already doing a lookup, no point on asking again"). Long term we should only trigger one lookup.